### PR TITLE
CentOS CI: add job for validating the ci/centos branch

### PR DESCRIPTION
**Pull request by nixpanic**
This PR adds validation for MarkDown, shell-scripts and yaml files in the ci/centos branch when a PR is created/updated.

While working on this, many files needed corrections. These corrections are required for the new jobs to succeed.

After a merge in the ci/centos branch is done, the jjb-deploy job will deploy the contents to the CentOS CI so that next PRs will get tested with the new configuration/jobs.

### CODE REVIEW

**Comment by nixpanic** [comment]
@Yuggupta27 please review this. Thanks!

**Comment by Madhu-1** [comment]
@nixpanic can we get job link when the Job is completed/in-progress to check the status?

File: scripts/Dockerfile.test [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Container image for running the Ceph-CSI tests
```
Comment (Madhu-1) [inline scripts/Dockerfile.test:1]: LGTM. I have a question
why we need containerized tests in centos CI?
can we run all static tests in Travis CI and run only time-consuming tests ie E2E in centos CI?

**Comment by nixpanic** [comment]
> @nixpanic can we get job link when the Job is completed/in-progress to check the status?

Yes, that will be added (is it the `context` in the `.yaml` files)

File: scripts/Dockerfile.test (hunk 1 above)
Comment (nixpanic) [reply scripts/Dockerfile.test:1]: It is that, or installing all dependencies on the CICO_NODE. Using the container image is just easier.

Comment (nixpanic) [reply scripts/Dockerfile.test:1]: Also, changes to the container image in the master branch can easily be backported to the ci/centos branch. If there is a separate script, we need to maintain is differently.

**Comment by nixpanic** [comment]
/retest ci/centos/job-validation

Typo in the manual created test-job (trying to run a .yaml as .groovy)

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+lint-markdown:
+	./scripts/lint-extras.sh lint-markdown
+
+lint-yaml:
+	./scripts/lint-extras.sh lint-yaml
+
+commitlint:
+	commitlint --from $(GIT_SINCE)
+
+.PHONY: containerized-test
```
Comment (Madhu-1) [inline Makefile:59]: do we need this in the centos branch? or the CI can pull it from the master/release  branch.

File: .commitlintrc.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# - value
+rules:
+  # the subject may not exceed 72 characters
+  header-max-length: [2, always, 72]
+  # the subject may not end with a "."
+  header-full-stop: [2, never, "."]
+  # we do not use scopes for commit messages
+  scope-empty: [1, always]
+  # valid types/prefixes, see docs/development-guide.md
+  type-enum:
```
[the remark this answers is not in the document]
Comment (nixpanic) [reply .commitlintrc.yml:15]: we do not need everything, but keeping the file in-sync with the one in the master branch makes it easier to backport changes

**Comment by nixpanic** [comment]
/test ci/centos/job-validation

@Yuggupta27 I needed to add a `--base=..` to `prepare.sh`, please review again.

**Comment by nixpanic** [comment]
:v: this now works, and it consumes the contents of the PR. @Yuggupta27 you can do something like that in your `e2e.yaml` too.

@Yuggupta27 can you also do a final check and approve again, as two commits have been added?

**Pull request merged**